src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,48 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        final LocalDate dateFromFormatted = LocalDate.parse(dateFrom, formatter);
+        final LocalDate dateToFormatted = LocalDate.parse(dateTo, formatter);
+        final int[] employeesSalaryCount = new int[names.length];
+        int countOfNames = 0;
+        StringBuilder reportBuilder = new StringBuilder();
+        reportBuilder.append("Report for period " + dateFrom + " - " + dateTo);
+        for (String name : names) {
+            for (String datum : data) {
```

Comment:
```suggestion
        StringBuilder reportBuilder = new StringBuilder();
        reportBuilder.append("Report for period " + dateFrom + " - " + dateTo);
        for (String name : names) {
            int salary = 0;
            for (String datum : data) {
```
you don't need that array